Codecademy/client-modules
# feat(Mobile Menu Variants): global header mobile

### Overview




cowritten by @katiezutter 

_Created the AppHeaderMobile_
Modified AppHeaderMainMenuMobile to be able to render the profile dropdown, and added the search component/functionality to it
Add variants for the different mobile headers (ex anonymous, free, pro)
Some code clean up
Updated stories + added mock data to be shared across the many stories requiring AppHeaderItems
Added render props for mobile search + notifications
Fixed mobile menu accessibility bug as well : https://codecademy.atlassian.net/browse/EGG-507 @jcenglish tagging you to take a look at the mobile menu from an a11y angle!  

*please note we have tickets for design check + documentation polishing


Testing Instructions!
1. Visit [preview link](https://6021be9f7574fa1cdcd2fd2a--gamut-preview.netlify.app/storybook/)
2. Check out the `AppHeaderMainMenu` [story](https://6021be9f7574fa1cdcd2fd2a--gamut-preview.netlify.app/storybook/?path=/docs/labs-experimental-molecules-appheadermainmenumobile--app-header-main-menu-mobile)
3. Check out the `AppHeaderMobile` [story](https://6021be9f7574fa1cdcd2fd2a--gamut-preview.netlify.app/storybook/?path=/docs/labs-experimental-molecules-appheadermobile--app-header-mobile)
4. Check out the `Global Navigation` [story](https://6021be9f7574fa1cdcd2fd2a--gamut-preview.netlify.app/storybook/?path=/docs/labs-experimental-molecules-globalheader--global-header) & toggle your screen between mobile & desktop and see that the header changes from the mobile version to the desktop version!
5. Check using voice over or by tabbing through that when the mobile menu is open, you cannot tab through or scroll through content behind the mobile menu

### PR Checklist

- [x] Related to JIRA ticket: [EGG-744
](https://codecademy.atlassian.net/browse/EGG-744)
